### PR TITLE
apps/examples/elf: Update Makefile to auto link libraries with wifi-micom apps

### PR DESCRIPTION
**Pull request by sangam-swami**
link available user libraries with micom and wifi applications

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [482846b574629f01028b8921d321ed8c79b227ce] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [482846b574629f01028b8921d321ed8c79b227ce] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@sangam-swami Could you make a new make stub for adding those kind change?
I want user just includes one file to do it. I don't want user should modify several lines in his own makefile.

**Comment by sunghan-chang** [comment]
@sangam-swami And there is no adding ld option for --gc-sections.

**Comment by sangam-swami** [comment]
> @sangam-swami Could you make a new make stub for adding those kind change?
> I want user just includes one file to do it. I don't want user should modify several lines in his own makefile.

Modified

**Comment by seinfra** [comment]
Target : [87da4d2a5f9c03c3396d6a7039c256a26bc7eb52] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [87da4d2a5f9c03c3396d6a7039c256a26bc7eb52] - Code Rule Check (C++) OK.

**Comment by sangam-swami** [comment]
> @sangam-swami And there is no adding ld option for --gc-sections.
With the current LDFLAGS apps don't have unused symbols, so I don't think we need this option

File: apps/AppLibs.mk [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+
+-include $(TOPDIR)/.config # Current configuration
+
+ifeq ($(CONFIG_BUILD_PROTECTED),y)
+include $(TOPDIR)/ProtectedLibs.mk
+else
+include $(TOPDIR)/FlatLibs.mk
```
Comment (sunghan-chang) [inline apps/AppLibs.mk]: This case is not necessary. There is no need to include this with FLAT BUILD. Right?

**Comment by seinfra** [comment]
Target : [04342a96c489c45289c4dc5dbff075edcf357d05] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [04342a96c489c45289c4dc5dbff075edcf357d05] - Code Rule Check (C++) OK.

File: apps/AppLibs.mk (hunk 1 above)
Comment (sangam-swami) [reply apps/AppLibs.mk]: agree, Removed the include for flat build.mk

**Pull request merged**